Gillian-C/lib/Chunk.ml

```diff
@@ -33,21 +53,55 @@ let to_string = function
   | Mint64 -> "int64"
   | Mfloat32 -> "float32"
   | Mfloat64 -> "float64"
-  | Many32 -> "any32"
-  | Many64 -> "any64"
+  | Mptr -> "ptr"
 
 let pp fmt chunk = Fmt.pf fmt "%s" (to_string chunk)
-let type_of = Compcert.AST.type_of_chunk
+
+let type_of = function
+  | Mint64 -> Compcert.AST.Tlong
+  | Mfloat32 -> Compcert.AST.Tsingle
+  | Mfloat64 -> Compcert.AST.Tfloat
+  | Mptr ->
+      if Compcert.Archi.ptr64 then Compcert.AST.Tany64 else Compcert.AST.Tany32
```

Comment:
I think the right thing to do here is to return `Tlong` when `ptr64` is true and `Tint` otherwise. That's previous behaviour, expected by the memory model